---
title: "class15.Rmd"
author: "Sara Herrera (PID:A59011948)"
date: "11/17/2021"
output: github_document
---

# 1. Bioconductor and DESeq2 setup

Today we look at data from a published RNA-seq experiment where airway smooth muscle cells were treated with dexamethasone, a synthetic glucocorticoid steroid with anti-inflammatory effects (Himes et al. 2014).

```{r}
# library(DESeq2)
```


# 2. Load the contData and colData

We need 2 things
- 1: count data
- 2: colData (the metadata that tells us about the design of the experiment)

```{r}
counts <- read.csv("airway_scaledcounts.csv", row.names=1)
metadata <-  read.csv("airway_metadata.csv")
```

```{r}
head(counts)
```

```{r}
head(metadata)
```

Side note:
Let's check the correspondence of the metadata and count data setup.

```{r}
metadata$id
```

```{r}
colnames(counts)
```

We can use the '==' to see if they are the same:

```{r}
all(metadata$id == colnames(counts))
```

# Differential Gene Expression: Compare control to treated

First we need to access all the 'control' columns in our counts data

```{r}
control.inds <- metadata$dex == "control"
metadata[ control.inds, ]
# This will only print out in the metadata whatever is control
# Or which(control.ids == TRUE)
# To just get the control IDs:
control.ids <- metadata[ control.inds, ]$id
```

Use these IDs to access just the control columns of our 'counts' data

```{r}
head(counts[ ,control.ids])
```

```{r}
# To calculate the mean of expression for each gene in the control condition:
control.mean <- rowMeans(counts[ ,control.ids])
head(control.mean)
```

Now we do the same for the treated:

```{r}
treated.inds <- metadata$dex == "treated"
metadata[ treated.inds, ]
# This will only print out in the metadata whatever is treated
# Or which(treated.ids == TRUE)
# To just get the treated IDs:
treated.ids <- metadata[ treated.inds, ]$id
```

```{r}
# To calculate the mean of expression for each gene in the treated condition:
treated.mean <- rowMeans(counts[ ,treated.ids])
head(treated.mean)
```

> Q. Number of genes?

There are `r nrow(counts)` rows/genes in this dataset

```{r}
nrow(counts)
```


We will combine our meancount data for bookkeeping purposes

```{r}
meancounts <- data.frame(control.mean, treated.mean)
colSums(meancounts)
```

## Compare the control and the treated

A quick plot of our progress so far

```{r}
plot(meancounts)
```

This suggests that we need to perform a log transformation. Let's plot on a log scale.

```{r}
plot(meancounts, log="xy")
```

We often use log transformation (log2) as they make life much easier...
Examples:

```{r}
log2(20/20)
log2(10/20)
log2(80/20)
```

```{r}
# The meancounts table has 2 columns: control.mean and treated.mean, and we're going to add a third column to this table, log2fc
meancounts$log2fc <- log2(meancounts[,"treated.mean"]/meancounts[,"control.mean"])
head(meancounts)
```

We need to drop the zeros and invalid lines!

```{r}
head(meancounts[,1:2] == 0)
```

The `which()` function tells us the indices of TRUE entries in a logical vector.
Example:

```{r}
which(c(T,F,T))
```

However, it's not that useful in default mode on our type of multi column input...

```{r}
# arr.ind will tell me the column and row position with 0's in it
inds <- which(meancounts[,1:2] == 0, arr.ind=TRUE)
head(inds)
```

I only care about the rows here (if there's a 0 in any column I will exclude this row eventually).

```{r}
to.rm <- unique(sort(inds[,"row"]))
```

```{r}
# To check, this command will show all the rows that have 0's to check with the id
head(meancounts[to.rm,])
```

```{r}
# To check the genes that we actually want to work with:
mycounts <- meancounts[-to.rm,]
head(meancounts[-to.rm,])
```

We now have `r nrow(mycounts)` genes remaining.

```{r}
nrow(mycounts)
```

How many of these genes are up regulated at the log2 fold-change threshold of +2 or greater?

```{r}
sum(mycounts$log2fc > 2)
```

What % is that?

```{r}
round((sum(mycounts$log2fc > 2) / nrow(mycounts)) * 100, 2)
```

How many of these genes are down regulated at the log2 fold-change threshold of -2 or smaller?

```{r}
sum(mycounts$log2fc < (-2))
round((sum(mycounts$log2fc < (-2)) / nrow(mycounts)) * 100, 2)
```

# DESeq 2 analysis

```{r}
library(DESeq2)
```

We first need to setup the DESeq input object:

```{r}
dds <- DESeqDataSetFromMatrix(countData=counts, 
                              colData=metadata, 
                              design=~dex)
```

Run the DESeq analysis pipeline.

```{r}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
head(res)
```

# Volcano Plot:

A very common data visualization plot.

```{r}
plot(res$log2FoldChange, -log(res$padj),
     xlab="Log2(FoldChange)",
     ylab="-Log(P-value)")
abline(v=c(-2,2), col="gray")
abline(h=-log(0.05), col="gray")
```

# 5. Add annotation data

Add meaningful gene names
```{r}
# Install packages first "AnnotationDbi" (does the work) and "org.Hs.eg.db" (contains the data that we're going to map between) through BiocManager::install()
library("AnnotationDbi")
library("org.Hs.eg.db")
```

```{r}
columns(org.Hs.eg.db)
# This has just information about species and other data in other websites
```
Here we map to "SYMBOL", the common gene name that everyone understands:

```{r}
res$symbol <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), # Our gene names
                     keytype="ENSEMBL",        # The format of our genenames
                     column="SYMBOL",          # The new format we want to add
                     multiVals="first")
head(res)
```

# To save our results:

```{r}
write.csv(res, file="allmyresults.csv")
```

# To make an enhanced volcano plot:

```{r}
library(EnhancedVolcano)
x <- as.data.frame(res)

EnhancedVolcano(x,
    lab = x$symbol,
    x = 'log2FoldChange',
    y = 'pvalue')
```

# 7. Pathway analysis

To bring some biology insight with pathway analysis.

```{r message=FALSE}
library(pathview)
library(gage)
library(gageData)
```

```{r}
data(kegg.sets.hs)

# Examine the first 2 pathways in this kegg set for humans
head(kegg.sets.hs, 2)
```

Before we can use KEGG we need to get our gene identifiers to the correct format for KEGG which is ENTREZ format (in this case).

```{r}
head(rownames(res))
```

```{r}
res$entrez <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), 
                     keytype="ENSEMBL",        
                     column="ENTREZID",          
                     multiVals="first")

res$genename <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), 
                     keytype="ENSEMBL",        
                     column="GENENAME",          
                     multiVals="first")
```

```{r}
head(res)
```

The main **gage()** function requires a named vector of fold changes, where the names of the values are the Entrez gene IDs.

Note that we used the mapIDs() function above to obtain Entrez gene IDs (stored in res$entrez) and we have the fold change results from DESeq2 analysis (stored in res$log2FoldChange).

```{r}
foldchanges = res$log2FoldChange
head(foldchanges)
```

Assign names to this vector that are the gene Ids that KEGG wants.

```{r}
# If we used res$symbol we would understand the gene names, but this is for KEGG
names(foldchanges) = res$entrez
```

```{r}
head(foldchanges)
```

Now we're ready to run **gage()**

```{r}
# Get the results
keggres = gage(foldchanges, gsets=kegg.sets.hs)
```

We can look at the attributes of an object:

```{r}
attributes(keggres)
```

```{r}
# Look at the first three down (less) pathways
head(keggres$less, 3)
```

To overlap our genes into the Asthma Pathway from KEGG we use **pathview()**:

```{r}
pathview(gene.data=foldchanges, pathway.id="hsa05310")
```

![](hsa05310.pathview.png)

You can also plot counts for specific genes of interest
















